pyresample/_formatting_html.py

```diff
@@ -0,0 +1,356 @@
+#!/usr/bin/env python
+#
+# Copyright (C) 2010-2020 Pyresample developers
+#
+# This program is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Functions for html representation of area definition."""
+
+import uuid
+from functools import lru_cache
+from html import escape
+from importlib.resources import read_binary
+
+import numpy as np
+
+import pyresample.geometry as geom
+
+try:
+    import cartopy
+    cart = True
+except ModuleNotFoundError:
+    cart = False
+
+try:
+    import xarray as xr
+    from xarray.core.formatting_html import _obj_repr, datavar_section
+    xarray = True
+except ModuleNotFoundError:
+    xarray = False
+
+
+STATIC_FILES = (
+    ("pyresample.static.html", "icons_svg_inline.html"),
+    ("pyresample.static.css", "style.css"),
+)
+
+
+@lru_cache(None)
+def _load_static_files():
+    """Lazily load the resource files into memory the first time they are needed."""
+    return [
+        read_binary(package, resource).decode("utf-8")
```

Comment:
Any idea if `read_binary` suffers from the same performance issues that `pkg_resources` utilities do when used in an environment with a lot of packages installed?

Reply:
Good question. I don't really know. This is probably something that I would move inline in a refactoring step similar to https://github.com/pytroll/satpy/pull/2171. That would also get rid of the directories which I think are not very nice.